qmail.c

```diff
@@ -23,29 +23,40 @@ struct qmail *qq;
 {
   int pim[2];
   int pie[2];
+  int pierr[2];
 
   setup_qqargs();
 
   if (pipe(pim) == -1) return -1;
   if (pipe(pie) == -1) { close(pim[0]); close(pim[1]); return -1; }
+  if (pipe(pierr) == -1) {
+    close(pim[0]); close(pim[1]);
+    close(pie[0]); close(pie[1]);
+    close(pierr[0]); close(pierr[1]);
```

Comment:
This isn't needed: the creation of these fds failed, so there is nothing to close.

Reply:
`pipe(2)` can't possibly half-succeed? For instance, create one, encounter a full file descriptor table, and fail the other?

Reply:
I don't think so, that wouldn't make any sense and can't be properly detected.